solutions/tests/README.md

```diff
@@ -1 +1,146 @@
-# Tests
+# Structured Test Solution and Key Testing Standards
+
+## Overview
+
+This document outlines the structure, guidelines, and standards for the test
+ written by our group for the ET code review project
+
+---
+
+## Table of Contents
+
+1. [Test Structure](#test-structure)
+2. [Testing Guidelines](#testing-guidelines)
+3. [Test Standards](#test-standards)
+4. [Test Coverage](#test-coverage)
+5. [Running Tests](#running-tests)
+
+---
+
+## Test Structure
+
+The test solutions are organized into the following directories:
+
+- **Unit Tests**:
+  - Located in the `/unit` directory.
+  - **Purpose**: Validates individual functions or methods in isolation,
+   ensuring they perform as expected.
+
+### Naming Conventions
+
+- **Test files**: Should follow the pattern `test_<module_name>.py`.
+- **Test functions**: Should be named `test_<functionality>`, e.g., `test_login`.
+
+---
+
+## Testing Guidelines
+
+### Writing Tests
+
+- **Isolate Tests**: Each test should be independent and should not rely on
+   shared states from other tests. This prevents flakiness.
+  
+- **Small Tests**: Focus on testing a specific functionality to make failure
+   easier to diagnose.
+  
+- **Descriptive Test Names**: Test names should describe exactly what is being
+   tested, for example, `test_user_login_with_valid_credentials`.
+
+---
+
+## Test Coverage
+
+- **Target Coverage**: Aim for high test coverage, but prioritize testing the
+   critical paths. Quality matters more than quantity—avoid excessive or
+  redundant tests.
+  
+- **Use Coverage Tools**: Regularly measure coverage using tools like
+   `coverage.py` or `pytest-cov` to ensure all edge cases, normal cases, and
+    error-handling scenarios are tested.
+
+### Test Dependencies
```

Comment:
Can you also delete the rest of this README because it is not particularly relevant to the project. Right?

Reply:
keep test coverage or remove it also?

Reply:
No, I think it makes sense and it is good to have it. Please who know it can use it.

Reply:
But you can decide whether to keep it or not.

Reply:
i am gonna push the edited here, and then if anyone else have any opinion we can talk about